### PR TITLE
Implement power spectrum normalization and include it in SVD pipeline closes #14

### LINKED ISSUE #14: Implement power spectrum normalization

### PR DESCRIPTION
**Pull request by DSilva27**
Included power spectrum normalization in `_preprocessing/normalize.py`. I included functions to compute power spectrum and also to normalize a set of volumes. Tried to use NUFFT for a better estimation of the power spectrum, but I couldn't manage to go back to real space. I think you are not supposed to go back anyway, as the adjoint transform is not necessarily the inverse transform. If you know how to revert NUFFT, then I would prefer to implement the normalization using that.

### CODE REVIEW

**Review by DavidHerreros — CHANGES_REQUESTED** [review]
It would be nice to add to the template svd_config.yaml file the new mandatory parameters (ref_vol_key, ref_vol_index, and output_file).

Also, having a short description in the SVD jupyter notebook of those new keys might be helpful for future users. Something like:

- ref_vol_key (str): the identifier of the submission where the reference volume needed to normalize the data will be taken (such as the icecream flavour of one of the submissions)
- ref_vol_index (int): in combination with ref_vol_key, determines the volume that will be set as the reference for the normalization of the data
- output_file (path): file where the ouputs will be saved. Must have .pt extension  -  (I think output_file substitutes output_path right?)

**Comment by DSilva27** [comment]
I implemented the changes. Also, I updated the config to make it more clear that the keys are related to power spectrum normalization.

**Comment by geoffwoollard** [comment]
@DSilva27 can you get the unit tests working (can be merged in from main), so I can see they pass in the PR?

**Comment by DSilva27** [comment]
Merged from main (actually, from 39-fix-small-bug-in-preprocessing-pipeline, so revise that one first). 

This PR does the following updates

**Major Changes**
- Implements power spectrum normalization using one submission as reference.

**Minor Changes**
- Fixes the script that downloads the testing data so it is all inside tests/data. Updates all tests so they are consistent with this change.
- Fixes tests generating files outside of tests/results

Closes #14 and #41

File: tests/config_files/test_config_map_to_map.yaml [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   ground_truth:
-    volumes: tests/data/Ground_truth/test_maps_gt_flat_10.pt 
-    metadata: tests/data/Ground_truth/test_metadata_10.csv 
-  mask: 
+    volumes: tests/data/Ground_truth/test_maps_gt_flat_10.pt
+    metadata: tests/data/Ground_truth/test_metadata_10.csv
+  mask:
     do: true
-    volume: data/Ground_truth/mask_dilated_wide_224x224.mrc
+    volume: tests/data/Ground_truth/mask_dilated_wide_224x224.mrc
```
Comment (geoffwoollard) [inline tests/config_files/test_config_map_to_map.yaml:14]: Remove tests from here, as it's the mask for the regular results. It agrees with the directories on OSF.

File: tests/scripts/fetch_test_data.sh [hunk 2]
```diff
@@ -3,9 +3,9 @@ wget https://files.osf.io/v1/resources/8h6fz/providers/dropbox/tests/dataset_2_s
...
 wget https://files.osf.io/v1/resources/8h6fz/providers/dropbox/tests/Ground_truth/test_maps_gt_flat_10.pt?download=true -O tests/data/Ground_truth/test_maps_gt_flat_10.pt
-wget https://files.osf.io/v1/resources/8h6fz/providers/dropbox/tests/Ground_truth/test_metadata_10.csv?download=true -O tests/data/Ground_truth/test_metadata_10.csv 
-wget https://files.osf.io/v1/resources/8h6fz/providers/dropbox/tests/Ground_truth/1.mrc?download=true -O tests/data/Ground_truth/1.mrc 
-wget https://files.osf.io/v1/resources/8h6fz/providers/dropbox/Ground_truth/mask_dilated_wide_224x224.mrc?download=true -O data/Ground_truth/mask_dilated_wide_224x224.mrc 
+wget https://files.osf.io/v1/resources/8h6fz/providers/dropbox/tests/Ground_truth/test_metadata_10.csv?download=true -O tests/data/Ground_truth/test_metadata_10.csv
+wget https://files.osf.io/v1/resources/8h6fz/providers/dropbox/tests/Ground_truth/1.mrc?download=true -O tests/data/Ground_truth/1.mrc
+wget https://files.osf.io/v1/resources/8h6fz/providers/dropbox/Ground_truth/mask_dilated_wide_224x224.mrc?download=true -O tests/data/Ground_truth/mask_dilated_wide_224x224.mrc
```
Comment (geoffwoollard) [inline tests/scripts/fetch_test_data.sh:8]: remove tests from here to be consistent.

File: tutorials/2_tutorial_svd.ipynb
Comment (geoffwoollard) [inline tutorials/2_tutorial_svd.ipynb:1]: Strongly suggested: Include rendered plots (low dpi) in this showing some results. People can compare against this when they reproduce things locally.

File: tutorials/3_tutorial_map2map.ipynb
Comment (geoffwoollard) [inline tutorials/3_tutorial_map2map.ipynb:1]: what did you change in this?

File: tutorials/4_tutorial_distribution2distribution.ipynb
Comment (geoffwoollard) [inline tutorials/4_tutorial_distribution2distribution.ipynb:1]: what did you change in this?

**Review by geoffwoollard — CHANGES_REQUESTED** [review]
Wow that's a lot of work, good job. Almost there.

I really think some SVD plots would help serve as documentation.

**Comment by geoffwoollard** [comment]
@DSilva27 There are some merge conflicts to attend to. But after that, since all the tests are passing, why don't you merge this in?

**Comment by DSilva27** [comment]
Making an overhaul of how the normalization is done. Will close this for now.

**Pull request closed without merging**